config/reader.go

```diff
@@ -498,7 +499,9 @@ func readFromCloud(
 		}
 	}
 
-	if prevCfg != nil && shouldCheckForCert(prevCfg.Cloud, cfg.Cloud) {
+	logger.Debugw("tls data", "cert", tlsCertificate, "key", tlsPrivateKey)
```

Comment:
Again I'm open to disagreement, but my knee-jerk reaction is never to log the private key, even at the debug level. Maybe you intend to remove this line again before the PR is merged?

Reply:
yikes, yeah this one I meant to remove - good catch

```suggestion
```